src/pages/academy/groundControl/subcomponents/GroundControlConfigureCell.tsx

```diff
@@ -117,12 +137,43 @@ const ConfigureCell: React.FC<Props> = ({ handleConfigureAssessment, data }) =>
                     label="Export Score Leaderboard (Coming soon!)"
                   />
                 </div>
-                <Switch
-                  className="publish-voting"
-                  disabled={true}
-                  inline
-                  label="Publish Voting (Coming soon!)"
-                />
+                <div className="current-voting-status">
+                  <InfoSign />
+                  <p className="voting-status-text">
+                    Current Voting Status: Entries have {!isVotingPublished && <b>not </b>} been
+                    assigned
+                  </p>
+                </div>
+                {!confirmAssignEntries ? (
+                  <div className="control-button-container">
+                    <ControlButton
+                      icon={IconNames.RESET}
+                      onClick={onAssignClick}
+                      label={`${!isVotingPublished ? 'Assign' : 'Reassign'} entries for voting`}
+                    />
+                  </div>
+                ) : (
+                  <div className="confirm-assign-voting">
+                    <Icon icon="reset" />
+                    <p className="confirm-assign-text">
+                      Are you sure you want to{' '}
+                      <b>{isVotingPublished ? 're-assign' : 'assign'} entries?</b>
+                    </p>
+                    <ButtonGroup>
+                      <Button small intent="success" onClick={handleConfirmAssign}>
+                        Assign
+                      </Button>
+                      <Button small intent="danger" onClick={handleCancelAssign}>
+                        Cancel
+                      </Button>
+                    </ButtonGroup>
+                  </div>
+                )}
+                {isVotingPublished && (
+                  <p className="reassign-voting-warning">
+                    <b>All existing votes</b> will be <b>deleted</b> upon reassigning entries!
+                  </p>
+                )}
```

Comment:
Thanks for grouping the state logic together using comments. Perhaps it could be better if we can have a separate sub-subcomponent for this toggle instead for better abstraction? I can imagine once we add more configurations into this cell it can become more complex – the `GroundControlConfigureCell` just be focused on the layout, ideally.

Reply:
On that note, do you think it might be a good idea to abstract out the other switches as well (i.e. team assessment, is publish voting, etc.?) I know they're pretty small components but they still might clutter up the code once you add more and more stuff to the config panel

Reply:
I was thinking about it, and I came to the conclusion that we should just abstract as necessary (i.e. only this component, at least for now) – what do you think?

Reply:
Okay then, sounds good. I'll refactor just the assign entries portion inside a new file.